### PR TITLE
feat: #187 add gap property

### LINKED ISSUE #187: Flex gap support RN 0.71.X
Flex gap RN 0.71.X, still not supported yet

### PR DESCRIPTION
**Pull request by mlecoq**
Resolves #187

I have upgraded the lib to support new gap property of RN 0.71 

I had to make some adjustments on types to fix typescript issues

### CODE REVIEW

**Comment by mlecoq** [comment]
cla signed

**Comment by mattfrances** [comment]
@mlecoq thanks for doing this! To isolate the changes of this PR to just be related to the gap property, I'll be making a PR upgrading this package's React Native version to 0.71. Once that's done, you'll be able to rebase and push only the changes specific to the gap property. The upgrade PR should be up within the next few days, thanks for your patience!

**Review by fortmarek — COMMENTED** [review]
This looks good!

I know we haven't done this for the other properties since it _is_ somewhat redundant to test that the individual props are passed to `style` based on the `true` flag in `restyleFunction`. But I'd still really love if we could get some basic coverage at least for new props. That is there should be some test that fails before implementing the rest of the PR.

I suggest adding a test like this to `createRestyleComponent.test.tsx`:
```tsx
    it('uses gap values from the theme', () => {
      const {root} = render(
        <ThemeProvider theme={theme}>
          <Component gap="s" columnGap="s" rowGap="s" />
        </ThemeProvider>,
      );
      expect(root.findByType(View).props).toStrictEqual({
        style: [{gap: 8, columnGap: 8, rowGap: 8}],
      });
    });
```

Additionally, let's also add an entry into `CHANGELOG` 🙏

**Comment by fortmarek** [comment]
@mlecoq could you also update the `README` with the docs for the gap property? Afterwards, LGTM 🚀

**Comment by mlecoq** [comment]
> @mlecoq could you also update the `README` with the docs for the gap property? Afterwards, LGTM 🚀

I have added gap properties to the list + updated an example where `columnGap` looks more appropriated than `marginRight`

**Comment by mlecoq** [comment]
You can use a patch with patch-package if you need it now

You just have to add 3 lines in src/restyleFunctions.ts

```
gap:true,
columnGap: true,
rowGap: true
```

**Review by fortmarek — APPROVED** [review]
Thanks a lot for your contribution ❤️ Let's ship it!
We'll do a release soon with this one 🙂

File: src/restyleFunctions.ts [hunk 1]
```diff
@@ -44,6 +47,9 @@ const spacingPropertiesShorthand = {
   py: 'paddingVertical',
   ps: 'paddingStart',
   pe: 'paddingEnd',
+  g: 'gap',
+  rG: 'rowGap',
+  cG: 'columnGap',
```
Comment (matinzd) [inline src/restyleFunctions.ts:52]: Wasn't it better to keep abbreviations consistent?`rg` and `cg`?

Comment (mlecoq) [reply src/restyleFunctions.ts:52]: You're right, I don't use shorthands and didn't notice they are in lowercase

**Comment by hamdij0maa** [comment]
@fortmarek please any idea about adding : the cornerCurve prop ? added in RN 0.71.X

**Comment by hamdij0maa** [comment]
@mlecoq  yes hhh

**Comment by mlecoq** [comment]
I'll make a PR

**Comment by mlecoq** [comment]
it's missing in react native types, have to make a PR on types repo first

**Comment by hamdij0maa** [comment]
yes, it works only on iOS also I think

**Pull request merged**